File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.java
class Solution {
    public int findLHS(int[] nums) {
        Arrays.sort(nums);
        int numsSize = nums.length;
        int j = 0;
        int maxLen = 0;
        for(int i = 0 ; i < numsSize ; i+=1){
            while(nums[i] - nums[j] > 1){
                j++;
            }

            if(nums[i] - nums[j] == 1){
                maxLen = Math.max(maxLen, i - j +1);
            }

        }
        return maxLen;

    }
}

File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        mpp = {}
        for i in nums :
            mpp[i] =  mpp.get(i, 0)+1
        res = 0
        for key in mpp.keys():
            if key + 1 in mpp:
                res = max(res, mpp.get(key) + mpp.get(key + 1))
        return res